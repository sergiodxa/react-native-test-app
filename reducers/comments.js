import { Map as map } from 'immutable';

function comments(state = map(), action = {}) {
  switch (action.type) {
    case 'ADD_COMMENT': {
      return state.set(action.payload.id, map(action.payload));
    }
    default: {
      return state;
    }
  }
}

export default comments;
