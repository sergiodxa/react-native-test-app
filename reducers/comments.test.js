import comments from './comments';

const action = {
  type: 'ADD_COMMENT',
  payload: {
    id: 1,
    postId: 1,
    name: 'id labore ex et quam laborum',
    email: 'Eliseo@gardner.biz',
    body: 'laudantium enim quasi est quidem magnam voluptate ipsam eos tempora quo necessitatibus dolor quam autem quasi reiciendis et nam sapiente accusantium',
  },
};

describe('comments reducer', () => {
  it('should add a new comment', () => {
    const state = comments(undefined, action);
    expect(state).toMatchSnapshot();
  });
});
