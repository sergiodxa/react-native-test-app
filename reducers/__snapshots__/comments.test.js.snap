// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comments reducer should add a new comment 1`] = `
Object {
  "1": Object {
    "body": "laudantium enim quasi est quidem magnam voluptate ipsam eos tempora quo necessitatibus dolor quam autem quasi reiciendis et nam sapiente accusantium",
    "email": "Eliseo@gardner.biz",
    "id": 1,
    "name": "id labore ex et quam laborum",
    "postId": 1,
  },
}
`;
